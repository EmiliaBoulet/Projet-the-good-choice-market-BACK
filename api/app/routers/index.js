//* Import de Router de express
import Router from "express";
const indexRouter = Router();

import userRouter from "./userRouter.js";
indexRouter.use("/api", userRouter);

import tvaRouter from "./tvaRouter.js";
indexRouter.use("/api", tvaRouter);

import productRouter from "./productRouter.js";
indexRouter.use("/api", productRouter);

import companyRouter from "./companyRouter.js";
indexRouter.use("/api", companyRouter);

import categoryRouter from "./categoryRouter.js";
indexRouter.use("/api", categoryRouter);

import brandRouter from "./brandRouter.js";
indexRouter.use("/api", brandRouter);

import imageRouter from "./imageRouter.js";
indexRouter.use("/api", imageRouter);

import errorRouter from "./errorRouter.js";
indexRouter.use("*", errorRouter);

export default indexRouter;
