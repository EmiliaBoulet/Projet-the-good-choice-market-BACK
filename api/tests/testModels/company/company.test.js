import request from "supertest";
import app from "../../appTest.js";

describe("GET ALL /company", () => {
    it("responds with json", async () => {
        const response = await request(app)
            .get("/api/company")
            .set("Accept", "application/json")
            .expect("Content-Type", /json/)
            .expect(200);
        expect(response.statusCode).toBe(200);

        for (const oneCompany of response.body) {
            expect(oneCompany).toMatchSnapshot({
                name: expect.stringMatching(/[a-zA-Z]/),
                phone_number: expect.stringMatching(/[0-9]{8,9}/)
            });
        }
    });
});
