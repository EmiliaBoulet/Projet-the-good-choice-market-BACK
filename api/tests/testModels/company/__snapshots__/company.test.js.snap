// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET ALL /company responds with json 1`] = `
Object {
  "address": "12 avenue de la banane plantain",
  "city": "Fort de France",
  "created_at": "2022-08-05T14:25:17.071Z",
  "email": "compagniecreole@plantain.com",
  "id": 1,
  "name": StringMatching /\\[a-zA-Z\\]/,
  "phone_number": StringMatching /\\[0-9\\]\\{8,9\\}/,
  "siret_number": "2598423675412",
  "updated_at": null,
}
`;

exports[`GET ALL /company responds with json 2`] = `
Object {
  "address": "94 boulevard des helvetiques",
  "city": "Bern",
  "created_at": "2022-08-05T14:25:17.071Z",
  "email": "contact@compagniedemeubles.ch",
  "id": 2,
  "name": StringMatching /\\[a-zA-Z\\]/,
  "phone_number": StringMatching /\\[0-9\\]\\{8,9\\}/,
  "siret_number": "2945832514789",
  "updated_at": null,
}
`;

exports[`GET ALL /company responds with json 3`] = `
Object {
  "address": "15 Rue Rivoli",
  "city": "Paris",
  "created_at": "2022-08-05T14:25:17.071Z",
  "email": "contact-us@compa-ny.com",
  "id": 3,
  "name": StringMatching /\\[a-zA-Z\\]/,
  "phone_number": StringMatching /\\[0-9\\]\\{8,9\\}/,
  "siret_number": "2589636658412",
  "updated_at": null,
}
`;
